internal/api/submit_download.go

```diff
@@ -0,0 +1,91 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/opg-sirius-finance-admin/internal/model"
+	"net/http"
+)
+
+func (c *ApiClient) SubmitDownload(ctx Context, reportType string, reportJournalType string, reportScheduleType string, reportAccountType string, reportDebtType string, dateField string, dateFromField string, dateToField string, emailField string) error {
+	var body bytes.Buffer
+	var dateTransformed *model.Date
+	var toDateTransformed *model.Date
+	var fromDateTransformed *model.Date
+
+	if dateField != "" {
+		raisedDateFormatted := model.NewDate(dateField)
+		dateTransformed = &raisedDateFormatted
+	}
+
+	if dateToField != "" {
+		startDateFormatted := model.NewDate(dateToField)
+		toDateTransformed = &startDateFormatted
+	}
+
+	if dateFromField != "" {
+		endDateFormatted := model.NewDate(dateFromField)
+		fromDateTransformed = &endDateFormatted
+	}
+
+	err := json.NewEncoder(&body).Encode(model.Download{
+		ReportType:         reportType,
+		ReportJournalType:  reportJournalType,
+		ReportScheduleType: reportScheduleType,
+		ReportAccountType:  reportAccountType,
+		ReportDebtType:     reportDebtType,
+		DateField:          dateTransformed,
+		ToDateField:        toDateTransformed,
+		FromDateField:      fromDateTransformed,
+		Email:              emailField,
+	})
+	if err != nil {
+		return err
+	}
+
+	req, err := c.newSiriusRequest(ctx, http.MethodPost, "/downloads", &body)
```

Comment:
Nothing is going to be posted to Sirius. The only thing Sirius will be used for is auth.

Reply:
Changed